vaadin-list-box-flow-parent/vaadin-list-box-flow/src/main/java/com/vaadin/flow/component/listbox/MultiSelectListBox.java

```diff
@@ -66,8 +66,16 @@ public MultiSelectListBox() {
     private static <T> JsonArray modelToPresentation(
             MultiSelectListBox<T> listBox, Set<T> model) {
         JsonArray array = Json.createArray();
-        model.stream().map(listBox.getItems()::indexOf)
-                .forEach(index -> array.set(array.length(), index));
+        model.stream()
+                .filter(selectedItem -> listBox.getItems()
+                        .stream()
+                        .anyMatch(item ->
+                                listBox.getItemId(selectedItem)
+                                        .equals(listBox.getItemId(item))))
+                .forEach(itm -> {
+                    int idx = listBox.getItems().indexOf(itm);
```

Comment:
`indexOf` is based on equals(), so it should be replaced somehow to use the custom identifier, as `listBox.getItemId` does.

Reply:
Done.